packages/lodestar/src/network/attestation-service/service.ts

```diff
@@ -0,0 +1,384 @@
+          this.subscribeToNextForkRandomSubnets(nextFork.name);
+        }, timeToPreparedEpoch);
+      }
+    }
+  }
+
+  private getNextFork(): {name: ForkName; slot: Slot} | null {
+    const currentFork = this.chain.getForkName();
+    const forkInfoRecord = this.config.getForkInfoRecord();
+    const allForks = Object.keys(forkInfoRecord) as ForkName[];
+    const forkIndex = allForks.indexOf(currentFork);
+    // there is a planned hard fork
+    if (forkIndex !== allForks.length - 1) {
+      const nextFork = allForks[forkIndex + 1];
+      return forkInfoRecord[nextFork];
+    }
+    return null;
+  }
+
+  /**
+   * Add validator to knownValidators list.
+   */
+  private addKnownValidator(validatorIndex: ValidatorIndex): void {
+    const currentSlot = this.chain.clock.currentSlot;
+    const currentFork = this.chain.getForkName();
+    if (!this.knownValidators.get(validatorIndex)) {
+      if (this.randomSubnets.getActive(currentSlot).length < ATTESTATION_SUBNET_COUNT) {
+        for (let i = 0; i < this.config.params.RANDOM_SUBNETS_PER_VALIDATOR; i++) {
+          this.subscribeToRandomSubnet(currentFork);
+        }
+      }
+    }
+    this.knownValidators.set(validatorIndex, currentSlot + LAST_SEEN_VALIDATOR_TIMEOUT);
+  }
+
+  /**
+   * Subscribe to long-lived random subnets and update the local ENR bitfield.
+   */
+  private subscribeToRandomSubnet(fork: ForkName): void {
+    const currentSlot = this.chain.clock.currentSlot;
+    const allSubnets = Array.from({length: ATTESTATION_SUBNET_COUNT}, (_, i) => i);
+    const activeSubnets = this.randomSubnets.getActive(currentSlot);
+    const availableSubnets = allSubnets.filter((subnet) => !activeSubnets.includes(subnet));
+    const toSubscribeSubnet = availableSubnets[randBetween(0, availableSubnets.length)];
+    // the heartbeat will help connect to respective peers
+    this.randomSubnets.request({
+      subnetId: toSubscribeSubnet,
+      toSlot: getSubscriptionSlotForRandomSubnet(this.config, currentSlot),
+    });
+    if (!this.subscribedCommitteeSubnets.getActive(currentSlot).includes(toSubscribeSubnet)) {
+      // subscribe to topic
+      this.gossip.subscribeTopic({type: GossipType.beacon_attestation, fork, subnet: toSubscribeSubnet});
+    }
+    // Update ENR
+    const attnets = this.metadata.attnets;
+    if (!attnets[toSubscribeSubnet]) {
+      attnets[toSubscribeSubnet] = true;
+      this.metadata.attnets = attnets;
+    }
+  }
+
+  /**
+   * Duplicate randomSubnets to nextForkRandomSubnets to make sure same ENR advertisement
+   * Subscribe to respective gossip topics with next fork.
+   * No need to update ENR
+   * @param nextFork
+   */
+  private subscribeToNextForkRandomSubnets(nextFork: ForkName): void {
+    const currentSlot = this.chain.clock.currentSlot;
+    const activeSubnets = this.randomSubnets.getActive(currentSlot);
+    for (const subnet of activeSubnets) {
+      this.nextForkRandomSubnets!.request({
```

Comment:
Could you find a way to not have to use the `!` operator? Maybe ensure it's defined first of change the data structure